src/App.tsx

```diff
@@ -1,37 +1,63 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
-  return `Clock-${value}`;
-}
+export class App extends React.Component<State> {
+  nameId = 0;
+
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  handleNameChange = () => {
+    this.nameId = window.setInterval(() => {
+      const newName = Date.now().toString().slice(-4);
+
+      this.setState({ clockName: `Clock-${newName}` });
+    }, 3300);
+  };
+
+  handleClockHide = (event: MouseEvent) => {
+    event.preventDefault(); // not to show the context menu
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+    this.setState({ hasClock: false });
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  handleClockShow = () => {
+    this.setState({ hasClock: true });
+  };
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidMount(): void {
+    document.addEventListener('contextmenu', this.handleClockHide);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', this.handleClockShow);
 
-      <div className="Clock">
-        <strong className="Clock__name">{clockName}</strong>
+    this.handleNameChange();
+  }
 
-        {' time is '}
+  componentWillUnmount(): void {
+    document.removeEventListener('contextmenu', this.handleClockHide);
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    document.removeEventListener('click', this.handleClockShow);
+
+    clearInterval(this.nameId);
+  }
+
+  render() {
+    const { hasClock } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {hasClock && <Clock name={this.state.clockName} />}
```

Comment:
You can also destructure clockName
```suggestion
        {hasClock && <Clock name={clockName} />}
```